Design/DesignDoc.md

```diff
@@ -0,0 +1,225 @@
+
+* Displaying expenditure and budget information in a graphical representation 
+* Rolling over remaining budget information for the next term 
+* Calculating balance remaining in budget after expenses  
+
+
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+This initial iteration will provide the minimum viable product (MVP) including adding, retrieving, and updating an expense information for the user role.
+
+We will use API Gateway and Lambda to create 6 endpoints (GetExpense, Add/UpdateExpense, GetTaggedExpense, GetBudget, Add/UpdateBudget, BudgetStatus) that will handle the creation, update, and retrieval of expense and budget information to satisfy our requirements.
+
+We will store expense and budget information in a table in DynamoDB.
+
+Amazon Elite Budget Service will also provide a web interface for users. A main page providing a variety of options to add, update and view expense and budget information.
+
+
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ExpenseModel
+
+String expenseId; (Generated UUID or 5/6 digit unique code)
+String expenseName;
+String expenseAmount; 
+String tag;
+ZonedDateTime expenseDate;
+
+// BudgetModel
+
+String budgetId; (Generated UUID or 5/6 digit unique code)
+String targetAmount; 
+Boolean status; 
+ZonedDateTime date;
+
+```
+
+## 6.2. _GetExpense Endpoint_
+
+* Accepts `GET` requests to `/expense/:expenseId`
+* Accepts an expenseID and returns the corresponding ExpenseModel.
+    * If the given expenseID is not found, will throw a
+      `ExpenditureNotFoundException`
+
+## 6.3 _GetAllExpenses Endpoint_
+
+* Accepts `GET` requests to `/expense`
+* Returns all the expenses with any tag in the ExpenseModel format.
+    * If there is no data found, will throw a
+      `NoDataFoundException`
+
+## 6.4. GetTaggedExpense Endpoint
+
+* Accepts `GET` requests to `/expense/tags/:tag`
+* Returns all the expenses for the requested tag in the ExpenseModel format.
+    * If there is no department found, will throw a
+      `InvalidDepartmentException`
+
+## 6.5. _UpdateExpense Endpoint_
+
+* Accepts `PUT` requests to `/expense/:expenseId`
+* Accepts data to update an expense including an updated
```

Comment:
Please list the data shape here (and for all other PUT/POST endpoints) like `{"expenseName": "name", "expenseAmount": 100, ...}`